Source/WebCore/css/StyleProperties.cpp

```diff
@@ -967,64 +1046,73 @@ static bool canOmitTrailingGridAreaValue(CSSValue& value, CSSValue& trailing)
     return isValueID(trailing, CSSValueAuto);
 }
 
-String StyleProperties::getGridRowColumnShorthandValue(const StylePropertyShorthand& shorthand) const
+String ShorthandSerializer::serializeGridRowColumn() const
 {
-    ASSERT(shorthand.length() == 2);
-    auto start = getPropertyCSSValue(shorthand.properties()[0]);
-    auto end = getPropertyCSSValue(shorthand.properties()[1]);
-    if (canOmitTrailingGridAreaValue(*start, *end))
-        return start->cssText();
-    return makeString(start->cssText(), " / ", end->cssText());
+    return serializeLonghands(canOmitTrailingGridAreaValue(longhandValue(0), longhandValue(1)) ? 1 : 2, " / ");
```

Comment:
I think I would keep the assert, in case a new longhand is added in the future.
Ditto for `serializeGridArea`.

Reply:
I am not sure those assertions are so great. I agree that they will start failing if we add another longhand to these shorthands but I am not sure a failing assertion is all that helpful. The incorrect serialization seems like the best way to notice the code is incorrect.

But it's easy for me to leave them in.

Reply:
My reasoning is that by failing they will attract attention to the the method that needs to have its logic updated. Just checking if the serialization is correct has the problem that not all combinations can be tested, so it may be missed.

Reply:
I think runtime assertions are pretty weak for this purpose, and checking just the length of the shorthand in particular is even weaker, misses things like if we reorder the properties in the shorthand. Instead we should probably switch to static assertions so the serialization code will immediately fail to compile if we change the shorthands. And we should figure out what exactly we are trying to assert. Might be nice to literally repeat the properties in the shorthand right there in the function in an assertion rather than just a comment.